src/anomalib/data/datamodules/base/image.py

```diff
@@ -81,8 +93,12 @@ def __init__(
         train_batch_size: int,
         eval_batch_size: int,
         num_workers: int,
-        val_split_mode: ValSplitMode | str,
-        val_split_ratio: float,
+        train_augmentations: Transform | None = None,
+        val_augmentations: Transform | None = None,
+        test_augmentations: Transform | None = None,
+        augmentations: Transform | None = None,
```

Comment:
Looking at the naming, are we 100% clear that `augmentations` is the best naming in this case? The reason for asking is the annotation for `augmentations` is, obviously, `Transform`. 

Would it make absolute sense to users?

Reply:
Ok so there are transforms that can be pre-defined model specific and they are stored in the exported model. 
And there are augmentations that are only applied during training.
Both can be defined separately for training, testing and validation. 


Maybe it would make sense to call the transforms something like `input_image_transforms` ? Or something that makes clearer that it is "baked" into the model and the last thing that transforms images before being processed by the model..

Reply:
On the datamodule side, I think `augmentations` is a more accurate and descriptive name than `transforms`. The `augmentations` naming signals that these are meant to define the data augmentation pipeline. `transforms` is more ambiguous and can refer to both data augmentations and model-specific input image pre-processing.

On the model/pre-processor side, we could think of a more descriptive name than `transform`, maybe`input_transforms`?  In any case we should ensure that there is a thorough how-to guide that clearly explains these concepts.

---

> Both can be defined separately for training, testing and validation.

@alexriedel1  Please note that in the latest version of this PR, we only keep 1 transform on the `PreProcessor` side. Since the model always requires the inputs in the same format, we figured there was no longer a use-case for having separate model transforms between the different subsets (but please let us know if you disagree).

Reply:
> @alexriedel1 Please note that in the latest version of this PR, we only keep 1 transform on the `PreProcessor` side. Since the model always requires the inputs in the same format, we figured there was no longer a use-case for having separate model transforms between the different subsets (but please let us know if you disagree).

Yes thats a good idea!